File: code/java/FactSetPrices/v1/src/main/java/com/factset/sdk/FactSetPrices/models/RollingPeriod.java
/*
 * FactSet Prices API
 * Gain access to comprehensive global coverage for Equities & Fixed Income. Perform quick analytics by controlling the date ranges, currencies, and rolling periods, or simply request Open, High, Low, and Close prices. Easily connect pricing data with other core company data or alternative content sets using FactSet's hub and spoke symbology. <p>Equity and Fund Security types include Common Stock, ADR, GDR, Preferred, Closed-ended Fund, Exchange Traded Fund, Unit, Open-ended Fund, Exchange Traded Fund UVI, Exchange Traded Fund NAV, Preferred Equity, Non-Voting Depositary Receipt/Certificate, Alien/Foreign, Structured Product, and Temporary Instruments. Reference over 180,000+ active and inactive securities.</p><p>Fixed Income Security Types include Corporate Bonds, Treasury and Agency bonds, Government Bonds, and Municipals.</p><p><b>Rate limit is set to 25 requests per second</b>.</p> 
 *
 * The version of the OpenAPI document: 1.3.0
 * Contact: api@factset.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


package com.factset.sdk.FactSetPrices.models;

import java.util.Objects;
import java.util.Arrays;
import java.util.Map;
import java.util.HashMap;
import io.swagger.annotations.ApiModel;
import java.io.Serializable;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;
import com.factset.sdk.FactSetPrices.JSON;


import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

/**
 * Period of measure for the rolling cumulative return. This does not change display &#x60;frequency&#x60; but rather the underlying return calculation period. All periods are referencing actual periods of measure, not period-ends. For example, 1M rolling period will go back to that date the previous month, which is not always the month-end date.
 */
public enum RollingPeriod {
  
  _1D("1D"),
  
  _1W("1W"),
  
  _1M("1M"),
  
  _3M("3M"),
  
  _6M("6M"),
  
  _52W("52W"),
  
  _2Y("2Y"),
  
  _3Y("3Y"),
  
  _5Y("5Y"),
  
  _10Y("10Y");

  private String value;

  RollingPeriod(String value) {
    this.value = value;
  }

  @JsonValue
  public String getValue() {
    return value;
  }

  @Override
  public String toString() {
    return String.valueOf(value);
  }

  @JsonCreator
  public static RollingPeriod fromValue(String value) {
    for (RollingPeriod b : RollingPeriod.values()) {
      if (b.value.equals(value)) {
        return b;
      }
    }
    throw new IllegalArgumentException("Unexpected value '" + value + "'");
  }
}

File: code/typescript/FactSetPrices/v1/src/model/RollingPeriod.js
/**
 * FactSet Prices API
 * Gain access to comprehensive global coverage for Equities & Fixed Income. Perform quick analytics by controlling the date ranges, currencies, and rolling periods, or simply request Open, High, Low, and Close prices. Easily connect pricing data with other core company data or alternative content sets using FactSet's hub and spoke symbology. <p>Equity and Fund Security types include Common Stock, ADR, GDR, Preferred, Closed-ended Fund, Exchange Traded Fund, Unit, Open-ended Fund, Exchange Traded Fund UVI, Exchange Traded Fund NAV, Preferred Equity, Non-Voting Depositary Receipt/Certificate, Alien/Foreign, Structured Product, and Temporary Instruments. Reference over 180,000+ active and inactive securities.</p><p>Fixed Income Security Types include Corporate Bonds, Treasury and Agency bonds, Government Bonds, and Municipals.</p><p><b>Rate limit is set to 25 requests per second</b>.</p> 
 *
 * The version of the OpenAPI document: 1.3.0
 * Contact: api@factset.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */

import ApiClient from '../ApiClient';
/**
* Enum class RollingPeriod.
* @enum {}
* @readonly
*/
export default class RollingPeriod {
    
        /**
         * value: "1D"
         * @const
         */
        "1D" = "1D";

    
        /**
         * value: "1W"
         * @const
         */
        "1W" = "1W";

    
        /**
         * value: "1M"
         * @const
         */
        "1M" = "1M";

    
        /**
         * value: "3M"
         * @const
         */
        "3M" = "3M";

    
        /**
         * value: "6M"
         * @const
         */
        "6M" = "6M";

    
        /**
         * value: "52W"
         * @const
         */
        "52W" = "52W";

    
        /**
         * value: "2Y"
         * @const
         */
        "2Y" = "2Y";

    
        /**
         * value: "3Y"
         * @const
         */
        "3Y" = "3Y";

    
        /**
         * value: "5Y"
         * @const
         */
        "5Y" = "5Y";

    
        /**
         * value: "10Y"
         * @const
         */
        "10Y" = "10Y";

    

    /**
    * Returns a <code>RollingPeriod</code> enum value from a Javascript object name.
    * @param {Object} data The plain JavaScript object containing the name of the enum value.
    * @return {module:model/RollingPeriod} The enum <code>RollingPeriod</code> value.
    */
    static constructFromObject(object) {
        return object;
    }
}

